saleor/graphql/product/types/products.py

```diff
@@ -263,6 +270,14 @@ def resolve_stocks(root: models.ProductVariant, info, country_code=None):
             return root.stocks.annotate_available_quantity().all()
         return root.stocks.annotate_available_quantity().for_country(country_code).all()
 
+    @staticmethod
+    def resolve_quantity_available(
+        root: models.ProductVariant, info, country_code=None
+    ):
+        if not country_code:
+            return get_max_available_quantity_for_customer(root)
+        return get_available_quantity_for_customer(root, country_code)
```

Comment:
I am wondering if it wouldn't be better to keep one method `get_available_quantity_for_customer` and there handle the situation when `country_code` is None (call there `get_max_available_quantity_for_customer`).

Reply:
From my perspective, it should be a separate function because of different approaches. If we want to get stocks for the country we start from stocks. If we want to get max availability in stocks it better to start from shipping zones. I can extract this if to function "resolve_quantity_available" if you think it more readable. Should I do it?

Reply:
Hm, but this approaches concerns implementation not what we want to achieve. When I call `get_available_quantity_for_customer` I just want to get `available_quantity` I don't care what is the approach on the implementation site.

Reply:
I agree with @IKarbowiak that from dev perspective, I would rather have one function and a one obvious way to use to get the public stock quantity of a variant. Good docstring should then describe how it works with the country code and what does the result mean if I don't provide the country code.